src/main/java/com/adaptris/core/varsub/VariableExpander.java

```diff
@@ -55,45 +55,51 @@ public VariableExpander(String prefix, String suffix) {
   }
 
   public Properties resolve(Properties input) throws CoreException {
-    Properties resolved = resolveCustom(input);
     Properties result = new Properties();
-    Properties sysProps = System.getProperties();
-    Properties environment = PropertyFileLoader.getEnvironment();
-
-    Set<String> variables = createVariableNames(resolved);
-    Set<String> sysPropVariables = createVariableNames(sysProps);
-    Set<String> environmentVariables = createVariableNames(environment);
-
-    for (String key : resolved.stringPropertyNames()) {
-      String value = resolved.getProperty(key);
-      // Loop through and sort out all the defined variables first.
-      while (containsVariable(value, variables)) {
-        value = expand(value, resolved);
-      }
-      // Now loop through and expand any system properties.
-      while (containsVariable(value, sysPropVariables)) {
-        value = expand(value, sysProps);
+    try {
+      Properties resolved = resolveCustom(input);
+      Properties sysProps = System.getProperties();
+      Properties environment = PropertyFileLoader.getEnvironment();
+      for (String key : resolved.stringPropertyNames()) {
+        String value = resolved.getProperty(key);
+        value = handleExpansion(key, value, resolved);
+        log.trace("Initial key [{}], value[{}]", key, value);
+        // Loop through and sort out all the defined variables first.
+        value = handleExpansion(key, value, resolved);
```

Comment:
After having a second look for my changes in the UI, I'm wondering why line 65 and line 68 are doing the same thing?

Reply:
Ok, that's definitely a cut and paste refactoring error... Will fix that now.